ballerina/http_request.bal

```diff
@@ -414,17 +415,34 @@ public class Request {
 
     # Sets a `json` as the payload. If the content-type header is not set then this method set content-type
     # headers with the default content-type, which is `application/json`. Any existing content-type can be
-    # overridden by passing the content-type as an optional parameter.
+    # overridden by passing the content-type as an optional parameter. If the given payload is a record type 
+    # with the `@jsondata:Name` annotation, the `jsondata:toJson` function processes the name and populates 
```

Comment:
Can't we just say that internally the records will be converted to json using the `jsondata:toJson` method?

Reply:
Please check updated there 
271e798